repo2docker/buildpacks/julia/julia_project.py

```diff
@@ -0,0 +1,140 @@
+"""Generates a Dockerfile based on an input matrix for Julia"""
+import os
+import toml
+from ..python import PythonBuildPack
+from .julia_semver import find_semver_match
+
+class JuliaProjectTomlBuildPack(PythonBuildPack):
+    """
+    Julia build pack which uses conda.
+    """
+
+    # ALL EXISTING JULIA VERSIONS
+    # Note that these must remain ordered, in order for the find_semver_match()
+    # function to behave correctly.
+    all_julias = [
+        "0.7.0",
+        "1.0.0", "1.0.1", "1.0.2", "1.0.3",
+        "1.1.0",
+    ]
+
+    @property
+    def julia_version(self):
+        default_julia_version = '1.1.0'
+        if os.path.exists(self.binder_path('JuliaProject.toml')):
+            project_toml = toml.load(self.binder_path('JuliaProject.toml'))
+        else:
+            project_toml = toml.load(self.binder_path('Project.toml'))
+        if 'compat' in project_toml:
+            if 'julia' in project_toml['compat']:
+                julia_version_str = project_toml['compat']['julia']
+
+                # For Project.toml files, install the latest julia version that
+                # satisfies the given semver.
+                julia_version = find_semver_match(julia_version_str, self.all_julias)
+                if julia_version is None:
+                    return _default_julia_version
```

Comment:
```suggestion
                    return default_julia_version
```
to match line 23

Reply:
There is a 
```
else:
  return julia_version
```
missing here I think

Reply:
Might be worth adding a test with a `Project.toml` that selects a non default version to stop this bug from coming back.

Reply:
Alright, I changed the structure of that code part. I also added a test that should exercise it.